src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/FakeRoleAssignmentsGenerator.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.definition.CachedCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.definition.CaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.definition.UserRole;
+import uk.gov.hmcts.ccd.domain.service.common.CaseAccessService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.data.casedetails.SecurityClassification.RESTRICTED;
+import static uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType.SPECIFIC;
+import static uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType.STANDARD;
+
+@Component
+@ConditionalOnProperty(name = "enable-attribute-based-access-control", havingValue = "true")
```

Comment:
in theory this is redundant because this code execution is already behind an if:
```
        if (applicationParams.getEnablePseudoRoleAssignmentsGeneration()) {
```